### PR TITLE
[css-grid-1] Add test for exclusion of cyclic-sizing-dependency items from baseline alignment.

### PR DESCRIPTION
**Pull request by fantasai**
https://github.com/w3c/csswg-drafts/issues/1365

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/21517)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/312275151)

File: css/css-grid/alignment/grid-baseline-align-cycles-001.html [hunk 1]
```diff
@@ -0,0 +1,100 @@
+<!DOCTYPE html>
```
Comment (mrego) [inline css/css-grid/alignment/grid-baseline-align-cycles-001.html]: Nit: I'd have added `<meta charset=utf-8>` but it's not really important anyway.

**Review by mrego — COMMENTED** [review]
Thanks for doing the tests.
I added some comments inline but most of them are nits that you can just ignore. 😇

File: css/css-grid/alignment/references/grid-baseline-align-cycles-001-ref.html [hunk 2]
```diff
@@ -0,0 +1,89 @@
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (foolip) [reply css/css-grid/alignment/references/grid-baseline-align-cycles-001-ref.html]: FWIW, I generally don't add it to keep tests as small as possible. https://web-platform-tests.org/writing-tests/general-guidelines.html is OK with that as long as the tests are pure ASCII.

File: css/css-grid/alignment/references/grid-baseline-align-cycles-001-ref.html [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+<title>
+	Reference
```
[the remark this answers is not in the document]
Comment (foolip) [reply css/css-grid/alignment/references/grid-baseline-align-cycles-001-ref.html]: There's a new GitHub feature where you can suggest edits directly. I'll make this edit though.

**Comment by foolip** [comment]
I've resolved some of the comments. @fantasai can you take a look at the remaining?

**Comment by zcorpan** [comment]
@mrego @fantasai or @tabatkins could one of you address the remaining comments, so this can land? Thanks!

**Comment by mrego** [comment]
I've applied my suggested changes as they were quite minor, but I'm not sure if I know how to rebase this thing.

BTW @javifernandez or @Loirooriol you might want to take a look into this test as it's not currently passing neither in Chromium or Firefox.

**Comment by zcorpan** [comment]
I've rebased now.

```
$ git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
$ git pull --prune
...
$ git remote add fantasai git@github.com:fantasai/web-platform-tests.git
$ git remote -v
fantasai	git@github.com:fantasai/web-platform-tests.git (fetch)
fantasai	git@github.com:fantasai/web-platform-tests.git (push)
origin	git@github.com:web-platform-tests/wpt.git (fetch)
origin	git@github.com:web-platform-tests/wpt.git (push)
$ git fetch fantasai
...
$ git checkout cyclic-baseline-items
Checking out files: 100% (41999/41999), done.
Branch 'cyclic-baseline-items' set up to track remote branch 'cyclic-baseline-items' from 'fantasai'.
Switched to a new branch 'cyclic-baseline-items'
$ git rebase origin/master
First, rewinding head to replay your work on top of it...
warning: unable to rmdir 'css/tools/apiclient': Directory not empty
warning: unable to rmdir 'css/tools/w3ctestlib': Directory not empty
Applying: [css-grid-1] Add test for exclusion of cyclic-sizing-dependency items from baseline alignment.
Applying: Update grid-baseline-align-cycles-001.html
Applying: Update grid-baseline-align-cycles-001-ref.html
Applying: Update grid-baseline-align-cycles-001.html
Applying: Applied some suggested improvements in grid-baseline-align-cycles-001.html test
$ git log
...looks good...
$ git push --force-with-lease --dry-run fantasai cyclic-baseline-items
...looks good...
$ git push --force-with-lease fantasai cyclic-baseline-items
```

**Comment by mrego** [comment]
Ok thanks, I guess I was not brave enough to use `--force-with-lease`; specially as it was not my own fork. :smile:

**Comment by zcorpan** [comment]
That's quite reasonable. :) An alternative would be to push to a new branch and open a new PR.

There are new [lint failures](https://community-tc.services.mozilla.com/tasks/Nhprq7nBQp6lGE_0dR57RQ/runs/0/logs/https%3A%2F%2Fcommunity-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FNhprq7nBQp6lGE_0dR57RQ%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log). @mrego do you think you can take a stab on fixing these?

**Comment by mrego** [comment]
lint failures should be gone now.

**Review by javifernandez — COMMENTED** [review]
The test looks good, but I think it'd better to define independent cases for the orthogonal item, since it will force a new pass of the track sizing algorithm, which may interfere with how the engine resolves the question of whether the grid area is intrinsically sized or not (see issue #3046)

**Comment by mrego** [comment]
> The test looks good, but I think it'd better to define independent cases for the orthogonal item, since it will force a new pass of the track sizing algorithm, which may interfere with how the engine resolves the question of whether the grid area is intrinsically sized or not (see issue #3046)

Thanks for the feedback @javifernandez, I think you have a good point in avoiding the "noise" introduced by the orthogonal item (which causes a 2nd pass of the grid track sizing algorithm); so I've separated the percent and orthogonal items in different grid containers, so orthogonal doesn't interfere on the final result of the percent item at all.

Since that's solved I guess this is ready to be merged.

**Review by javifernandez — APPROVED** [review]

**Pull request merged**